packages/editor-ui/src/views/NodeView.vue

```diff
@@ -2159,6 +2159,10 @@ export default mixins(
 						},
 					];
 
+					this.workflowsStore.addConnection({
```

Comment:
@OlegIvaniv as I remember the event connection callback that calls `workflowsStore.addConnection` is called also when you load a module and by passing `setStateDirty: true` the module immediately gets into dirty state and you get the unsaved change confirmation modal when you try to navigate away without editing the workflow.

Also `workflowsStore.addConnection` is called elsewhere in the process of connecting nodes and I thought it is unnecessary here.

Maybe this just should not change `setStateDirty` just use the value it has at the moment

Reply:
When we set up input endpoints we set target like this:
```
target: !this.isReadOnly && nodeTypeData.inputs.length > 1, // only enabled for nodes with multiple inputs.. otherwise attachment handled by connectionDrag event in NodeView,
```
That's why we can't rely only on `EVENT_CONNECTION_ABORT` hook to call `connectTwoNodes` which calls `this.workflowsStore.addConnection` because for nodes that have more than 1 input(merge node) we have to use `EVENT_CONNECTION` hook. So I've added this call to `EVENT_CONNECTION` and manually set `uiStore.stateIsDirty = true`